microcosm/object_graph.py

```diff
@@ -102,7 +103,7 @@ def __getattr__(self, key):
         return self._resolve_key(key)
 
     def __setattr__(self, key, value):
-        if not key.startswith("_") and key not in ("metadata", "config"):
+        if not key.startswith("_") and key not in ("metadata", "config", "loader"):
```

Comment:
I wanted to set this within creating an app from an actual service for testing first, but `setattr` is locked on the object graph here so I just decided to make the change I need in microcosm rather than writing a custom factory to save the `loader` on the graph.